File: Project/Project/frontend/src/services/certificateService.js
import http from './http';

export async function issue(body) {
  const res = await http.post('/certificates', body);
  return res.data.data;
}
export async function byCandidate(candidateId) {
  const res = await http.get(`/certificates/${candidateId}`);
  return res.data.data;
}
export async function expiring() {
  const res = await http.get('/certificates/expiring');
  return res.data.data;
}
export async function updateStatus(id, status) {
  const res = await http.patch(`/certificates/${id}/status`, { status });
  return res.data.data;
}

File: Project/Project/microservices/certificate-service/src/main/java/com/certifypro/certificate/service/CertificateService.java
package com.certifypro.certificate.service;

import com.certifypro.certificate.dto.request.IssueCertificateRequest;
import com.certifypro.certificate.dto.request.UpdateCertificateStatusRequest;
import com.certifypro.certificate.dto.response.CertificateResponse;

import java.util.List;

public interface CertificateService {

    /** Certification Officer / Admin manually issues a certificate. */
    CertificateResponse issue(IssueCertificateRequest req);

    /** Service-to-service auto-issue (called by result-service on Pass publish). */
    CertificateResponse issueInternal(Long candidateId, Long programId);

    /** Candidate views own certificates (officer/Admin may view any). */
    List<CertificateResponse> getByCandidate(Long candidateId);

    CertificateResponse updateStatus(Long id, UpdateCertificateStatusRequest req);

    /** Validity tracker: certificates expiring within the next 90 days. */
    List<CertificateResponse> expiringWithin90Days();
}
